scripts/install_backend_python_libs.py

```diff
@@ -0,0 +1,276 @@
+# Copyright 2020 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Installation script for Oppia python backend libraries."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import collections
+import os
+import shutil
+import subprocess
+
+from pip._internal.utils import misc
+import python_utils
+from scripts import common
+
+import pkg_resources
+
+
+def _get_requirements_file_contents():
+    """Returns a dictionary containing all of the required library names with
+    their corresponding version strings listed in the 'requirements.txt' file.
+
+    Returns:
+        dict(string, string). Dictionary with the name of the library as the key
+        and the version string of that library as the value.
+    """
+    requirements_contents = collections.defaultdict()
+    with python_utils.open_file(common.REQUIREMENTS_FILE_PATH, 'r') as f:
+        lines = f.readlines()
+        for l in lines:
+            if l.startswith('#') or len(l.strip()) == 0:
+                continue
+            library_and_version = l.split(' ')[0].split('==')
+            library = library_and_version[0].lower()
+            version = library_and_version[1]
+            requirements_contents[library] = version
+    return requirements_contents
+
+
+def _get_third_party_directory_contents():
+    """Returns a dictionary representing all of the libraries with their
+    corresponding versions installed in the 'third_party/python_libs' directory.
+
+    Returns:
+        dict(string, string). Dictionary with the name of the library installed
+        as the key and the version string of that library as the value.
+    """
+    installed_distributions = misc.get_installed_distributions(
```

Comment:
Not sure about this one, using internal pip libraries is not recommended (https://mail.python.org/pipermail/distutils-sig/2017-October/031642.html), `installed_packages = [(d.project_name, d.version) for d in pkg_resources.find_distributions(path)]` should work in a similar way (needs to be tested).